esmvalcore/preprocessor/_area.py

```diff
@@ -234,11 +247,11 @@ def area_statistics(cube, operator):
     try:
         grid_areas = cube.cell_measure('cell_area').core_data()
     except iris.exceptions.CellMeasureNotFoundError:
-        logger.info(
+        logger.debug(
             'Cell measure "cell_area" not found in cube %s. '
             'Check fx_file availability.', cube.summary(shorten=True)
         )
```

Comment:
This is already checked when parsing the recipe here: https://github.com/ESMValGroup/ESMValCore/blob/96518ed591510eca46b7402b9b60556f87ba1b96/esmvalcore/_recipe.py#L436-L438
and logs a warning much earlier than running the preprocessor. Repeating it several times coming from preprocessor functions is just making noise and you loose the original message because the screen fills with noise.

Reply:
yeahhh no - I think it's best we have it and redundancy is not a problem here since this is not much noise, it creeps up not very often and it needs big flashing lights if it happens. I'd really keep it in

Reply:
What the ACTUAL F***???? Do not move this message to debug. This is mission critical for me. Are you trying to give me a hernia?

Reply:
lol, sorry for the salty languge, I'm just a bit shocked. I realise that missing FX files don't always affect everyone as much as me, but this is one of the biggest problem areas in the CMIP6 datasets. So many models have misplaced or incorrectly labelled the Ofx data. 

Instead of a debug message, I would propose making this a FATAL error.  If you are missing FX data like areacello to perform a calculation (like Area mean or something), then you're unable to perform the calculation and ESMValTool should stop running.

Reply:
yeah but that means it'll go through a whole lot of time-consuming tasks before reaching that (incl. lots of data finding, other preprocessors etc etc). Especially if one runs single-threaded. Also, it might try using a file that the user doesn't want (depending how far we get with actually finding fx data when it's not really in the place specified in the recipe - see #1309 ) @ledm please edit your first comment, let's try and keep this decent since it's a public repo :beer:

Reply:
I don't understand why everyone is minimising the importance of fx files. If they are requested but absent, ESMValTool can not and should not proceed. We should not try to calculated the area, or somehow do without. We should not just go ahead and see where the preprocessor breaks.

In addition, the error message for when FX files are not found is opaque and hard to understand. It doesn't need to be changed, it just needs to not be called because ESMValTool should have stopped already, as soon as the fx files were found to be missing.

Reply:
@valeriupredoi 
> yeah but that means it'll go through a whole lot of time-consuming tasks before reaching that (incl. lots of data finding, other preprocessors etc etc).

What are you on about? This is all of 28 lines below this point. There is no data finding or other pre-processors in between, but rather only code paths that can not be taken if the error condition applies.

@ledm
As everyone knows, I am sympathetic to insisting on standards compliance. I also don't want to minimize FX files (though I think we can clean up how we speak about these fields because I think this is really more about cell measures even if they are not FiXed, rather than fixed fields like [(Efx, clayfrac)](http://clipc-services.ceda.ac.uk/dreq/u/81963e3a-f906-11e6-a176-5404a60d96b5.html)).

The *only* case in which ESMValTool will calculate areas is for regular latitude-longitude grids. Imho, this can be done reliably, and moreover, these areas are really often not published. In all other cases ESMValTool already behaves the way that you describe.

Reply:
I don't think we're calculating areas any more, but I would still argue that we shouldn't calculate it at all for CMIP6 models (Observations is a different and messier story).

Basically, because the Earth isn't perfectly round, an iris-calculated area in a regular grid may not match what was used at the model runtime. Different models can use represent the Earth using spherical or non-spherical globes, and that can make (albeit small) differences in area, even in the regular grid area.

I'm just arguing that I want the option for ESMValTool to stop if FX files are absent, and not to find out about their absence in an obscure error message after half an hour of preprocessing.

Reply:
> I don't think we're calculating areas any more, but I would still argue that we shouldn't calculate it at all for CMIP6 models (Observations is a different and messier story).

You think we are not calculating areas, but also we shouldn't do it. So all good?

> Basically, because the Earth isn't perfectly round, an iris-calculated area in a regular grid may not match what was used at the model runtime. Different models can use represent the Earth using spherical or non-spherical globes, and that can make (albeit small) differences in area, even in the regular grid area.

Yes. And if the models follow the standard and provide the necessary information as described [here](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.9/cf-conventions.html#grid-mappings-and-projections), this will be taken into account.

> I'm just arguing that I want the option for ESMValTool to stop if FX files are absent, and not to find out about their absence in an obscure error message after half an hour of preprocessing.

As is already the case. So, great, we are all good?

Reply:
Thank you for explaining @ledm. I propose we move this discussion to #1282, because stopping the run based on these missing files is a bit separate from the discussion on log messages. Note that, as I already mentioned earlier in this thread, there is a WARNING message that the fx files are missing issued when parsing the recipe, way before starting any preprocessing. It looks like this:
> 2021-09-14 09:39:53,142 UTC [14396] WARNING Missing data for fx variable areacella of dataset FGOALS-s2

and this pull request makes that more specific, so it looks like:

> 2021-09-14 09:39:53,142 UTC [14396] WARNING Missing data for fx variable 'areacella' of dataset CMIP5 FGOALS-s2 r1i1p1

Therefore I think it is really not needed to issue other messages coming from the preprocessor, because these will make the screen so full of messages that the original warning message is lost from view.

Reply:
Yeah, happy to move on. Sorry I derailed the discussion! 🚂